purchase_sale_stock_inter_company/tests/test_inter_company_purchase_sale_stock.py

```diff
@@ -0,0 +1,623 @@
+        )
+        purchase.order_line += purchase.order_line.copy({"product_qty": 2})
+        sale = self._approve_po(purchase)
+        sale.action_confirm()
+
+        # validate the SO picking
+        so_picking_id = sale.picking_ids
+
+        # Link to a new purchase order so it can trigger
+        # `PO does not exist or has no receipts` in _sync_receipt_with_delivery
+        sale.auto_purchase_order_id = purchase_2
+
+        # Set quantities done on the picking and validate
+        for move in so_picking_id.move_ids:
+            move.quantity = move.product_uom_qty
+        with self.assertRaises(UserError):
+            so_picking_id.button_validate()
+
+    def test_sync_picking_multi_step(self):
+        self.company_a.sync_picking = True
+        self.warehouse_a.reception_steps = "two_steps"
+        self.company_b.sync_picking = True
+        self.warehouse_c.delivery_steps = "pick_ship"
+
+        purchase = self._create_purchase_order(
+            self.partner_company_b, self.consumable_product
+        )
+        sale = self._approve_po(purchase)
+
+        self.assertEqual(len(purchase.picking_ids), 1)
+        # Only a single picking is created for the sale.
+        # When this picking is validated, two pickings should be created:
+        # one for the backorder and one for the delivery.
+        self.assertEqual(len(sale.picking_ids), 1)
+        # validate the SO internal picking
+        so_internal_pick = sale.picking_ids
+
+        so_internal_pick.move_ids.quantity = 2
+        so_internal_pick.move_ids.picked = True
+        wizard_data = so_internal_pick.with_user(self.user_company_b).button_validate()
+        wizard = (
+            self.env["stock.backorder.confirmation"]
+            .with_context(**wizard_data.get("context"))
+            .create({})
+        )
+        wizard.process()
+        po_picking = purchase.picking_ids
+        # check po_picking state
+        self.assertEqual(po_picking.state, "waiting")
+
+        # validate the SO picking
+        so_picking = sale.picking_ids.filtered(
+            lambda x: x.location_dest_id.usage == "customer"
+        )
+        so_picking.move_ids.quantity = 2
+        so_picking.move_ids.picked = True
+        self.assertNotEqual(po_picking, so_picking)
+        self.assertNotEqual(
+            po_picking.move_ids.quantity,
+            so_picking.move_ids.quantity,
+        )
+
+        so_picking.with_user(self.user_company_b).button_validate()
+        self.assertEqual(so_picking.state, "done")
+        po_internal_pick_id = po_picking.move_ids.move_dest_ids.picking_id
+        # the move in the receipt should have a "next move" due to "two_steps"
+        self.assertTrue(purchase.picking_ids.move_ids.move_dest_ids)
+        self.assertEqual(len(sale.picking_ids), 3)  # Pick + Backorder + Delivery
+        self.assertTrue(
+            all((po_picking, po_internal_pick_id, so_picking, so_internal_pick))
+        )
```

Comment:
```suggestion
        po_internal_pick = po_picking.move_ids.move_dest_ids.picking_id
        # the move in the receipt should have a "next move" due to "two_steps"
        self.assertTrue(purchase.picking_ids.move_ids.move_dest_ids)
        self.assertEqual(len(sale.picking_ids), 3)  # Pick + Backorder + Delivery
        self.assertTrue(
            all((po_picking, po_internal_pick, so_picking, so_internal_pick))
        )
```
Only to keep same policy with names in this test, but other tests are using "_id" so not blocking

Reply:
Done, thanks